### PR TITLE
feat: add arbitrum

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Add arbitrum support:
- Send/Receive
- Balances/TxHistory
- Nft Support
- Swap Support (zrx, oneinch, and lifi)

TODO:
- ~Support any other swappers with arbitrum support (cow, lifi, one inch)?~
- ~Update chain adapter and unchained client tests once unchained coinstack is up and running~
- ~~Bump hdwallet once merged and published~~

Follow Up:
- Fiat on-ramps

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Low

## Testing

- Test send and receive of native assets and tokens
- Validate correct balances, charts, and transaction history
- Validate correct nft display
- Test swapping native -> token, token -> native, token -> token
- Spot check functionality working on all supported wallets
- Spot check gas estimations/prices align with what is reported on block explorer

### Engineering

:point_up: 

### Operations

:point_up: 

## Screenshots (if applicable)

image

image

image

image

image

* Zrx:
image

* LiFi:
image

* OneInch:
image

### CODE REVIEW

File: packages/caip/src/adapters/coingecko/generated/eip155_42161/adapter.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"eip155:42161/slip44:60":"ethereum"}
```
Comment (kaladinlight) [inline packages/caip/src/adapters/coingecko/generated/eip155_42161/adapter.json:1]: where muh tokens?

File: packages/chain-adapters/src/evm/arbitrum/ArbitrumChainAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import * as unchained from '@shapeshiftoss/unchained-client'
+
+import { ChainAdapterDisplayName } from '../../types'
+import type { ChainAdapterArgs } from '../EvmBaseAdapter'
+import { EvmBaseAdapter } from '../EvmBaseAdapter'
+
+const SUPPORTED_CHAIN_IDS = [KnownChainIds.ArbitrumMainnet]
+const DEFAULT_CHAIN_ID = KnownChainIds.ArbitrumMainnet
+
+export class ChainAdapter extends EvmBaseAdapter<KnownChainIds.ArbitrumMainnet> {
```
Comment (kaladinlight) [inline packages/chain-adapters/src/evm/arbitrum/ArbitrumChainAdapter.ts:14]: validate if l1 fees are needed similar to optimism

Comment (kaladinlight) [reply packages/chain-adapters/src/evm/arbitrum/ArbitrumChainAdapter.ts:14]: estimations seem accurate without any additional l1 logic so far...

File: .env.base [hunk 3]
```diff
@@ -84,11 +85,14 @@ REACT_APP_ALCHEMY_API_KEY=3DnVf4ZoxBk0W0TZcipqAhmATm7uc1vE
 REACT_APP_ALCHEMY_POLYGON_JAYPEGS_API_KEY=OqVQS8M_NzYeV--9xHZjKEIscrZOADzp
 REACT_APP_ALCHEMY_ETHEREUM_JAYPEGS_API_KEY=yh6GF21GaU8wlvEB4wYp7ssXhCU76gOx
 REACT_APP_ALCHEMY_OPTIMISM_JAYPEGS_API_KEY=K03vX5rMZeGmF2V9U-SCsrCeeVnr8qmO
+REACT_APP_ALCHEMY_ARBITRUM_JAYPEGS_API_KEY=DqH3kvkDIq8qwit_G8TM4DFQe5s5TKdq
```
Comment (kaladinlight) [inline .env.base:88]: do we have a dao alchemy account that we are managing api keys from?

File: package.json [hunk 4]
```diff
@@ -93,17 +93,17 @@
...
+    "@shapeshiftoss/hdwallet-core": "^1.51.0",
+    "@shapeshiftoss/hdwallet-keepkey": "^1.51.0",
+    "@shapeshiftoss/hdwallet-keepkey-webusb": "^1.51.0",
+    "@shapeshiftoss/hdwallet-keplr": "^1.51.0",
+    "@shapeshiftoss/hdwallet-metamask": "^1.51.0",
+    "@shapeshiftoss/hdwallet-native": "^1.51.0",
+    "@shapeshiftoss/hdwallet-native-vault": "^1.51.0",
+    "@shapeshiftoss/hdwallet-shapeshift-multichain": "^1.51.0",
+    "@shapeshiftoss/hdwallet-walletconnectv2": "^1.51.0",
+    "@shapeshiftoss/hdwallet-xdefi": "^1.51.0",
```
Comment (gomesalexandre) [inline package.json:106]: 💜

File: packages/chain-adapters/src/evm/arbitrum/ArbitrumChainAdapter.test.ts [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+const makeGetAccountMockResponse = (balance: {
+  balance: string
+  tokenBalance: string | undefined
+}) => ({
+  balance: balance.balance,
+  unconfirmedBalance: '0',
+  nonce: 2,
+  tokens: [
+    {
+      assetId: `eip155:137/erc20:${contractAddress}`,
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/chain-adapters/src/evm/arbitrum/ArbitrumChainAdapter.test.ts:74]: good catch, thanks

**Review by gomesalexandre — COMMENTED** [review]
LGTM, conceptual stamp pending runtime testing

**Pull request merged**